pkg/test/masteruserrecord/master_user_record.go

```diff
@@ -1,70 +1,155 @@
 package masteruserrecord
 
 import (
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
 	"fmt"
 	"strings"
+	"testing"
 	"time"
 
 	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
 	"github.com/codeready-toolchain/toolchain-common/pkg/condition"
 	"github.com/codeready-toolchain/toolchain-common/pkg/test"
 	"github.com/redhat-cop/operator-utils/pkg/util"
 	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/require"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 )
 
-type MurModifier func(mur *toolchainv1alpha1.MasterUserRecord)
+type MurModifier func(mur *toolchainv1alpha1.MasterUserRecord) error
 type UaInMurModifier func(targetCluster string, mur *toolchainv1alpha1.MasterUserRecord)
 
-func NewMasterUserRecord(userName string, modifiers ...MurModifier) *toolchainv1alpha1.MasterUserRecord {
-	userId := uuid.NewV4().String()
+// DefaultNSTemplateTier the default NSTemplateTier used to initialize the MasterUserRecord
+var DefaultNSTemplateTier = toolchainv1alpha1.NSTemplateTier{
+	ObjectMeta: metav1.ObjectMeta{
+		Namespace: test.HostOperatorNs,
+		Name:      "basic",
+	},
+	Spec: toolchainv1alpha1.NSTemplateTierSpec{
+		Namespaces: []toolchainv1alpha1.NSTemplateTierNamespace{
+			{
+				TemplateRef: "basic-dev-123abc",
+			},
+			{
+				TemplateRef: "basic-code-123abc",
+			},
+			{
+				TemplateRef: "basic-stage-123abc",
+			},
+		},
+		ClusterResources: &toolchainv1alpha1.NSTemplateTierClusterResources{
+			TemplateRef: "basic-clusterresources-654321a",
+		},
+	},
+}
+
+// DefaultNSTemplateSet the default NSTemplateSet used to initialize the MasterUserRecord
+var DefaultNSTemplateSet = toolchainv1alpha1.NSTemplateSet{
+	ObjectMeta: metav1.ObjectMeta{
+		Namespace: test.HostOperatorNs,
+		Name:      DefaultNSTemplateTier.Name,
+	},
+	Spec: toolchainv1alpha1.NSTemplateSetSpec{
+		TierName: DefaultNSTemplateTier.Name,
+		Namespaces: []toolchainv1alpha1.NSTemplateSetNamespace{
+			{
+				TemplateRef: "basic-dev-123abc",
+			},
+			{
+				TemplateRef: "basic-code-123abc",
+			},
+			{
+				TemplateRef: "basic-stage-123abc",
+			},
+		},
+		ClusterResources: &toolchainv1alpha1.NSTemplateSetClusterResources{
+			TemplateRef: "basic-clusterresources-654321a",
+		},
+	},
+}
+
+func NewMasterUserRecords(t *testing.T, size int, nameFmt string, modifiers ...MurModifier) []runtime.Object {
+	murs := make([]runtime.Object, size)
+	for i := 0; i < size; i++ {
+		murs[i] = NewMasterUserRecord(t, fmt.Sprintf(nameFmt, i), modifiers...)
+	}
+	return murs
+}
+
+func NewMasterUserRecord(t *testing.T, userName string, modifiers ...MurModifier) *toolchainv1alpha1.MasterUserRecord {
+	userID := uuid.NewV4().String()
+	hash, _ := computeTemplateRefsHash(DefaultNSTemplateTier) // we can assume the JSON marshalling will always work
```

Comment:
if we can ignore the error here, why don't we ignore the error also in `Account`? I know that this is predefined but still, it looks weird. 
In addition, we have `*testing.T` instance here, so we can easily check the error also in this function.
Just to be clear - I don't wanna force you to do it (the probability of the error is really low), I just have a bad feeling when we ignore an error somewhere - that's why I shared my thought/concern.

Reply:
I believe I ignored the error at first for the sake for simplicity (ie, not returning an error, especially for test data), and I only added the `t *testing` param afterwards, so yes, I can to the checking now

Reply:
done in 97d17cc